pkg/service/cluster_test.go

```diff
@@ -88,12 +88,54 @@ func TestCreateCluster(t *testing.T) {
 	assert.Equal(t, *newCluster.Annotations, *cluster.Annotations)
 }
 
+var createClusterWithIDTests = map[string]struct{
+	request api.Id
+	response api.Id
+}{
+	"requested ID gets accepted": {
+		"c-my-custom-id",
+		"c-my-custom-id",
+	},
+	"ID without prefix gets prefixed": {
+		"my-custom-id",
+		"c-my-custom-id",
```

Comment:
I'd suggest to set the properties explicitly
```suggestion
		request: "c-my-custom-id",
		response: "c-my-custom-id",
	},
	"ID without prefix gets prefixed": {
		request: "my-custom-id",
		response: "c-my-custom-id",
```